hls4ml/writer/quartus_writer.py

```diff
@@ -277,72 +374,146 @@ def write_test_bench(self, model):
-                newline += '      inputs.emplace_back();\n'
-                for inp in model.get_input_variables():
-                    newline += f'      in_end = in_begin + ({inp.size_cpp()});\n'
-                    newline += f'      std::copy(in_begin, in_end, inputs.back().{inp.member_name});\n'
-                    newline += '      in_begin = in_end;\n'
-                newline += '      outputs.emplace_back();\n'
-            elif '//hls-fpga-machine-learning insert zero' in line:
-                newline = line
-                newline += indent + 'for(int i = 0; i < num_iterations; i++) {\n'
-                for inp in model.get_input_variables():
-                    newline += indent + f'  inputs.emplace_back();\n'
-                    newline += indent + f'  outputs.emplace_back();\n'
-                    newline += indent + f'  std::fill_n(inputs[i].{inp.member_name}, {inp.size_cpp()}, 0.0);\n'
-                newline += indent + '}\n'
-
-            elif '//hls-fpga-machine-learning insert top-level-function' in line:
-                newline = line
-
-                newline += indent + 'for(int i = 0; i < num_iterations; i++) {\n'
-                newline += indent + f'  ihc_hls_enqueue(&outputs[i], {model.config.get_project_name()}, inputs[i]);\n'
-                newline += indent + '}\n'
-            elif 'hls-fpga-machine-learning insert run' in line:
-                newline = line
-                newline += '    ' + 'ihc_hls_component_run_all({});\n'.format(model.config.get_project_name())
-            elif '//hls-fpga-machine-learning insert predictions' in line:
-                newline = line
-                newline += indent + 'for(int i = 0; i < {}; i++) {{\n'.format(outvar.size_cpp())
-                newline += indent + '  std::cout << predictions[j][i] << " ";\n'
-                newline += indent + '}\n'
-                newline += indent + 'std::cout << std::endl;\n'
-            elif '//hls-fpga-machine-learning insert tb-output' in line:
-                newline = line
-                newline += indent + 'for(int i = 0; i < {}; i++) {{\n'.format(outvar.size_cpp())
-                newline += indent + '  fout << outputs[j].{}[i] << " ";\n'.format(outvar.member_name)
-                newline += indent + '}\n'
-                newline += indent + 'fout << std::endl;\n'
-            elif '//hls-fpga-machine-learning insert output' in line or '//hls-fpga-machine-learning insert quantized' in line:
-                newline = line
-                newline += indent + 'for(int i = 0; i < {}; i++) {{\n'.format(outvar.size_cpp())
-                newline += indent + '  std::cout << outputs[j].{}[i] << " ";\n'.format(outvar.member_name)
-                newline += indent + '}\n'
-                newline += indent + 'std::cout << std::endl;\n'
-            else:
-                newline = line
-            fout.write(newline)
+        if io_type == 'io_parallel':
```

Comment:
Maybe split this giant if block into two sub-functions for brevity?

Reply:
Done.